api/src/main/java/org/openmrs/DrugOrder.java

```diff
@@ -512,13 +517,41 @@ public void setDosing(DosingInstructions di) {
 	 */
 	@Override
 	public boolean hasSameOrderableAs(Order otherOrder) {
-		if (!super.hasSameOrderableAs(otherOrder)) {
-			return false;
-		}
-		if (!(otherOrder instanceof DrugOrder)) {
-			return false;
-		}
-		DrugOrder otherDrugOrder = (DrugOrder) otherOrder;
-		return OpenmrsUtil.nullSafeEquals(this.getDrug(), otherDrugOrder.getDrug());
+        if (!super.hasSameOrderableAs(otherOrder)) {
+            return false;
+        }
+        if (!(otherOrder instanceof DrugOrder)) {
+            return false;
+        }
+        DrugOrder otherDrugOrder = (DrugOrder) otherOrder;
+
+        if (isNonCodedDrug() || otherDrugOrder.isNonCodedDrug()) {
+            return OpenmrsUtil.nullSafeEqualsIgnoreCase(this.getDrugNonCoded(), otherDrugOrder.getDrugNonCoded());
+        }
+        return OpenmrsUtil.nullSafeEquals(this.getDrug(), otherDrugOrder.getDrug());
+    }
+
+	/**
+	 * @since 1.12
+	 * @return drugNonCoded
+	 */
+	public String getDrugNonCoded() {
+		return drugNonCoded;
+	}
+
+	/**
+	 * @since 1.12
+	 * sets drugNonCoded
+	 */
+	public void setDrugNonCoded(String drugNonCoded) {
+		this.drugNonCoded = drugNonCoded != null ? drugNonCoded.trim() : drugNonCoded;
+	}
+
+	/**
+	 * @since 1.12
+	 * return true if a drug is non coded
+	 */
+	public boolean isNonCodedDrug() {
+		return getDrugNonCoded() != null;
```

Comment:
I think this should check that also the non coded value is not blank, we normally user StringUtils.isNotBlank